src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java

```diff
@@ -564,6 +568,9 @@ private ResponseEntity<PostePayRefundResponse> executeRefundRequest(String beare
             }
 
             requestEntity.setIsRefunded(refundOutcome.equals(EsitoStorno.OK));
+            if (requestEntity.getIsRefunded()){
```

Comment:
@andrearovere it's better if you avoid checking the entity soon after having set it.
You can refactor this as follows:

```
boolean isRefunded = refundOutcome.equals(EsitoStorno.OK);
if (isRefunded) {
     requestEntity.setStatus(PaymentRequestStatusEnum.CANCELLED.name());
}
requestEntity.setIsRefunded(isRefunded);
```

Also, is it correct to use enum status CANCELLED for refunded payment requests?
Maybe something like "REFUNDED" could be more appropriate: is this a requirement?

Reply:
Changed as suggested.
The approved states are CREATED; AUTHORIZED, DENIED and CANCELLED.